File: 0670-maximum-swap/0670-maximum-swap.py
class Solution:
    def maximumSwap(self, num: int) -> int:
        num=list(str(num))
        for i in range(0, len(num)):
            maxi=num[i]
            ind=i
            for j in range(i+1, len(num)):
                if ord(num[j])>=ord(maxi):
                    maxi=num[j]
                    ind=j
            if maxi!=num[i]:
                num[i], num[ind] = maxi, num[i]
                break
        return int("".join(num))

File: 0670-maximum-swap/0670-maximum-swap.java
class Solution {
    public int maximumSwap(int num) {
        char[] c = Integer.toString(num).toCharArray();
        // String[] ans = new String[c.length];
        char mx = c[c.length-1];
        int mxIdx = c.length-1;
        int ans = num;
        
        for(int i=c.length-1; i>=0; --i){
            char cur = c[i];
            if(cur > mx){
                mx = cur;
                mxIdx = i;
            }
            else{
                c[i] = mx;
                c[mxIdx] = cur;
                ans = Math.max(ans, Integer.parseInt(String.valueOf(c)));
                c[mxIdx] = mx;
                c[i] = cur;
            }
        }
        
        return ans;
    }
}